File: api/baeuja/src/main/java/xyz/baeuja/api/content/repository/ContentRepository.java
package xyz.baeuja.api.content.repository;

import org.springframework.data.domain.Pageable;
import org.springframework.data.domain.Slice;
import org.springframework.data.jpa.repository.JpaRepository;
import xyz.baeuja.api.content.domain.Classification;
import xyz.baeuja.api.content.domain.Content;

public interface ContentRepository extends JpaRepository<Content, Long> {

    Slice<Content> findAllByClassification(Classification classification, Pageable pageable);
}

File: android/baeuja/app/src/main/java/com/eello/baeuja/domain/content/repository/ContentRepository.kt
package com.eello.baeuja.domain.content.repository

import com.eello.baeuja.domain.common.pagination.model.Pagination
import com.eello.baeuja.domain.content.model.Classification
import com.eello.baeuja.domain.content.model.ContentMeta
import com.eello.baeuja.domain.content.model.ContentUnit
import com.eello.baeuja.domain.content.model.WordContent

interface ContentRepository {

    suspend fun getNewContents(): List<ContentMeta>
    suspend fun getWordContents(excludeWords: List<Int> = emptyList()): List<WordContent>
    suspend fun getMainContents(size: Int = 5): Map<Classification, List<ContentMeta>>
    suspend fun getContentsByClassification(
        classification: Classification,
        page: Int,
        size: Int = 5
    ): Pagination<ContentMeta>

    suspend fun getContentDetail(id: Long): ContentMeta

    suspend fun getContentUnits(contentId: Long, classification: Classification): List<ContentUnit>
}